### PR TITLE
Add membership model

### PR DESCRIPTION
**Pull request by cmutel**

### CODE REVIEW

**Comment by cmutel** [comment]
@lilliputten please merge if happy

File: dds_registration/models.py [hunk 1]
```diff
@@ -85,6 +85,24 @@ def __init__(self, *args, **kwargs):
         self._original_username = self.username
 
 
+class Membership(models.Model):
+    user = models.ForeignKey(User, related_name='registrations', on_delete=models.CASCADE)
+    started = models.IntegerField(default=lambda: date.today().year)
```
Comment (lilliputten) [inline dds_registration/models.py:90]: Is that ok to use this kind of period evaluation? Eg, if the user started his membership in December, it'll be ended in a month?

File: dds_registration/models.py [hunk 2]
```diff
@@ -85,6 +85,24 @@ def __init__(self, *args, **kwargs):
         self._original_username = self.username
 
 
+class Membership(models.Model):
```
Comment (lilliputten) [inline dds_registration/models.py:88]: Is there a migration required?

File: dds_registration/models.py [hunk 3]
```diff
@@ -85,6 +85,24 @@ def __init__(self, *args, **kwargs):
...
+    user = models.ForeignKey(User, related_name='registrations', on_delete=models.CASCADE)
+    started = models.IntegerField(default=lambda: date.today().year)
+    until = models.IntegerField(default=lambda: date.today().year)
```
Comment (lilliputten) [inline dds_registration/models.py:91]: While trying to create migrations, I'm getting errors for these lambda's:

`Cannot serialize function: lambda`.

File: dds_registration/models.py [hunk 4]
```diff
@@ -85,6 +85,24 @@ def __init__(self, *args, **kwargs):
         self._original_username = self.username
 
 
+class Membership(models.Model):
+    user = models.ForeignKey(User, related_name='registrations', on_delete=models.CASCADE)
```
Comment (lilliputten) [inline dds_registration/models.py:89]: I was asked to use a different related name. I've used `related_name='memberships'`.

File: dds_registration/models.py (hunk 2 above)
Comment (cmutel) [reply dds_registration/models.py:88]: Yes.

File: dds_registration/models.py (hunk 1 above)
Comment (cmutel) [reply dds_registration/models.py:90]: Yes, that's the way it works - that's why the text says "calendar year"

File: dds_registration/models.py (hunk 3 above)
Comment (cmutel) [reply dds_registration/models.py:91]: OK, let's switch to:

```python
def this_year():
    return date.today().year
```

And then `default=this_year`.

File: dds_registration/models.py (hunk 4 above)
Comment (cmutel) [reply dds_registration/models.py:89]: 👍🏽

**Pull request merged**